### PR TITLE
WSTEAMA 734 Lazy Load all homepage images apart from the first and message banner

### PR DESCRIPTION
**Pull request by holchris**
Resolves JIRA 734
https://jira.dev.bbc.co.uk/browse/WSTEAMA-734

Overall changes
======
Resolves Option 2 
Only the first image on the homepage and message banner is not lazy loaded, but all others are

Code changes
======

- Adds logic to Lazy Load all images on the Homepage, apart from the first image and message banner

Testing
======
- Adds unit test to check that all images apart from the first and message banner are not lazy loaded

### CODE REVIEW

**Review by karinathomasbbc — APPROVED** [review]
🐰

**Review by amoore108 — APPROVED** [review]
Nice!

File: src/app/pages/HomePage/index.test.tsx [hunk 1]
```diff
@@ -117,6 +121,37 @@ describe('Home Page', () => {
...
+        service: 'kyrgyz',
+      });
+
+      const imageList = document.querySelectorAll('img');
+      imageList.forEach((image, index) => {
+        if (index === 0) {
+          expect(image.getAttribute('loading')).toBeNull();
+        } else {
+          const src = image.getAttribute('src') || '';
+          if (messageBannerImages.includes(src)) {
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/index.test.tsx]: Should this logic be inverted? 
```suggestion
          if (!messageBannerImages.includes(src)) {
```

File: src/app/pages/HomePage/index.test.tsx [hunk 2]
```diff
@@ -117,6 +121,44 @@ describe('Home Page', () => {
...
+
+      const imageList = document.querySelectorAll('img');
+      imageList.forEach((image, index) => {
+        if (index === 0) {
+          expect(image.getAttribute('loading')).toBeNull();
+        } else {
+          const src = image.getAttribute('src') || '';
+          if (!messageBannerImages.includes(src)) {
+            expect(image.getAttribute('loading')).toBe('lazy');
+          }
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/index.test.tsx]: Can we tweak this slightly, to ensure that we test that the message banner images are not lazy loaded:

```suggestion
       const src = image.getAttribute('src') || '';

        if (index === 0 || messageBannerImages.includes(src)) {
          expect(image.getAttribute('loading')).toBeNull();
        } else  if (!messageBannerImages.includes(src)) {
            expect(image.getAttribute('loading')).toBe('lazy');
            }
```

Comment (karinathomasbbc) [reply src/app/pages/HomePage/index.test.tsx]: Maybe don't commit my suggestion, as I think the brackets are all messed up! Try this locally, and run the tests to ensure they are still passing.

File: src/app/pages/HomePage/index.test.tsx [hunk 3]
```diff
@@ -117,6 +117,43 @@ describe('Home Page', () => {
...
+    // const { BANNER } = VISUAL_STYLE;
+    // const { NORMAL } = VISUAL_PROMINENCE;
+    // const messageBannerImages = kyrgyzHomePageData.curations
+    //   .filter(
+    //     ({ visualStyle, visualProminence }) =>
+    //       visualStyle === BANNER && visualProminence === NORMAL,
+    //   )
+    //   .flatMap(curation => {
+    //     return curation.summaries?.map(summary => summary.imageUrl);
+    //   });
```
Comment (karinathomasbbc) [inline src/app/pages/HomePage/index.test.tsx]: Let's delete this now.

```suggestion
```

**Review by karinathomasbbc — APPROVED** [review]

**Pull request merged**